javatests/arcs/android/devtools/DevToolsStarterTest.kt

```
@@ -0,0 +1,112 @@
+  private lateinit var devToolsIntent: Intent
+
+  @Before
+  fun setUp() {
+    context = ApplicationProvider.getApplicationContext<Application>()
+    devToolsIntent = Intent(context, DevToolsService::class.java)
+    app = ApplicationProvider.getApplicationContext()
+  }
+
+  @Test
+  fun start_createsIntent() = lifecycle { _, serviceShadow ->
+    val devToolsStarter = DevToolsStarter(context)
+    val bundle = Bundle().apply {
+      putSerializable(STORAGE_CLASS, StorageService::class.java)
+    }
+    devToolsIntent.putExtras(bundle)
+
+    devToolsStarter.start()
+    val actualIntent = serviceShadow.nextStartedService
+
+    assertThat(actualIntent.filterEquals(devToolsIntent)).isTrue()
+    assertThat(actualIntent.extras?.getSerializable(STORAGE_CLASS))
+      .isEqualTo(devToolsIntent.extras?.getSerializable(STORAGE_CLASS))
+  }
+
+  @Test
+  fun start_withStorageService_appliesBundle() = lifecycle { _, serviceShadow ->
+      val devToolsStarter = DevToolsStarter(context)
+      val bundle = Bundle().apply {
+        putSerializable(STORAGE_CLASS, MySecondStorageService::class.java)
+      }
+      devToolsIntent.putExtras(bundle)
+
+      devToolsStarter.start(MySecondStorageService::class.java)
+      val actualIntent = serviceShadow.nextStartedService
+
+      assertThat(actualIntent.filterEquals(devToolsIntent)).isTrue()
+      assertThat(actualIntent.extras?.getSerializable(STORAGE_CLASS))
+        .isEqualTo(devToolsIntent.extras?.getSerializable(STORAGE_CLASS))
+    }
+
+  @Test
+  fun start_withBadStorageService_createsIntent() = lifecycle { _, serviceShadow ->
+    val devToolsStarter = DevToolsStarter(context)
+
+    devToolsStarter.start(MyClass::class.java)
+    val actualIntent = serviceShadow.nextStartedService
+
+    assertThat(actualIntent.filterEquals(devToolsIntent)).isTrue()
+    assertThat(actualIntent.extras?.getSerializable(STORAGE_CLASS))
+      .isEqualTo(devToolsIntent.extras?.getSerializable(STORAGE_CLASS))
+  }
```

This is a great test! It makes me think we may want to throw an exception if the user passes a non-StorageService subclass instead of silently using StorageService if what they pass is invalid. (Better to crash on unexpected input than give unexpected behavior)